persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/message/MessageDAO.kt

```diff
@@ -10,19 +10,27 @@ data class MessageEntity(
     val date: String,
     val senderUserId: QualifiedIDEntity,
     val senderClientId: String,
-    val status: Status
+    val status: Status,
+    val deleted: Deletion = Deletion.Neutral
 ) {
     enum class Status {
         PENDING, SENT, READ, FAILED
     }
+
+    enum class Deletion {
+        Neutral, DELETED, HIDDEN
+    }
 }
 
 interface MessageDAO {
     suspend fun deleteMessage(id: String, conversationsId: QualifiedIDEntity)
+    suspend fun deleteMessage(id: String)
+    suspend fun updateDeleteStatus(deleteStatus: MessageEntity.Deletion, content: String? = null, id: String)
```

Comment:
1. Consider adding a `conversationId` argument for extra safety like you have for `updateMessageStatus`
2. I'd remove the `content` argument. The DAO can clear the content field internally because soon content will not just be a string but it will depend on the message type (text, asset, ...).